Feature iops 2042 profile comparison nhs england profiles vs nhs digital profiles

KazeemHamzat opened the pull request:
Feature iops 2042 profile comparison nhs england profiles vs nhs digital profiles

KazeemHamzat committed:
Changes to England-Location-Organixzation-MedicationDispense

Changes to England-Location-Organixzation-MedicationDispense

KazeemHamzat committed:
Changes from Profile Comparisons

NHSD-Profiles vs NHSE Profiles

digitalhealthmartin commented on structuredefinitions/England-Location.xml:
According to spreadsheet, Location.type.extension:locationRole is missing max=1 but this isn't reflected here.

digitalhealthmartin replied:
Also, why is there only one slice in NHSE in Location.identifier but there are two in NHSD?

digitalhealthmartin replied:
And also the cardinality max in NHSD it * but you're proposing 1 in NHSE

KazeemHamzat replied:
1. max=1 has been added
2.  min=1, line 52 will resolve slicing issue
3.

KazeemHamzat replied:
max=1 changed to max=*

digitalhealthmartin commented on structuredefinitions/England-MedicationRequest.xml:
isModifier=false is there but in spreadsheet it states it should be mustSupport=true for the extensions

digitalhealthmartin replied:
On the spreadsheet it was raised about the min cardinality of MedicationRequest.subject.reference. It's currently min=1 in NHSE but it was not this in NHSD, why is this the case?

digitalhealthmartin replied:
On the spreadsheet it was raised about the min cardinality of MedicationRequest.dispenseRequest.performer.identifier. It's currently min=1 in NHSE but it was not this in NHSD, why is this the case?

digitalhealthmartin commented on structuredefinitions/England-ServiceRequest.xml:
```diff
@@ -205,10 +190,6 @@
       <sliceName value="SNOMEDProcedureCode" />
-      <binding>
-        <strength value="preferred" />
-        <valueSet value="https://fhir.hl7.org.uk/ValueSet/UKCore-ProcedureCode" />
```
Based on your discussions with UK Core, this was was advised?

KazeemHamzat replied:
Binding was removed to revert to present state in NHSDigital-ServiceRequest. 
Note: ServiceRequest.code.coding:SNOMEDProcedureCode has no binding. IOPS-2086 has been raised(needs DA discussion for resolution)

HamzatKazeem committed:
Resolving PR Feedback

Replacing MustSupport=True with isModifier=false in Extension, changing minimum cardinality to zero

digitalhealthmartin reviewed (changes requested):
I looked at my comments for Location and although you marked them as resolved and commented that you updated them I don't see that reflected in the code I don't think. I'll look at it again and the other profiles once you can show me the fixes are actually pushed in.

HamzatKazeem committed:
Update England-Location.xml

Location.type.extension:locationRole
max=1

SheenaSharma commented on structuredefinitions/England-Task.xml:
Checked with the changes as mentioned in the ticket (NHSDigital-Task vs England-Task) and it looks fine to me.

KazeemHamzat replied:
Resolution of England-ServiceRequest Ticket is not in this PR. The reason the status is still in-progress. 
The changes on England-ServiceRequest in this PR are Forge recommendations and a reversal to present state of NHSDigital-ServiceRequest.
England-ServiceRequest Ticket, [IOPS-2086](https://nhsd-jira.digital.nhs.uk/browse/IOPS-2086) has been scheduled for DA discussion and resolution.

KazeemHamzat commented:
> I looked at my comments for Location and although you marked them as resolved and commented that you updated them I don't see that reflected in the code I don't think. I'll look at it again and the other profiles once you can show me the fixes are actually pushed in.

Response to earlier comment on England-Location are in code lines 43 and 75.
1. set max=' * ' in Forge deletes the code.
2. set max= ' 1 '  for Location.type.extension:locationRole is now appearing.

SheenaSharma commented on structuredefinitions/England-ServiceRequest.xml:
ServiceRequest profile issues will be resolved/assigned in different feature branch.

SheenaSharma commented on structuredefinitions/England-PractitionerRole.xml:
Looks fine to me, as it doesn't have any major changes except removing <min value="1" />.

SheenaSharma commented on structuredefinitions/England-Practitioner.xml:
Looks fine, as it doesn't have any major changes except removing <min value="0" />.

SheenaSharma commented on structuredefinitions/England-Organization.xml:
Looks fine. Changes are done which are mentioned in the ticket.

SheenaSharma commented on structuredefinitions/England-OperationOutcome.xml:
Looks fine, changes are done as mentioned in the ticket.

SheenaSharma commented on structuredefinitions/England-MessageHeader.xml:
MessageHeader profile issues will be handled/resolved in different feature branch.

SheenaSharma commented on structuredefinitions/England-Patient.xml:
Forge recommendations, i.e. <min value="0" />are done.
Cardinality issues are to be treated by the UK Core Team.

SheenaSharma commented on structuredefinitions/England-Observation.xml:
Changes are done which are mentioned in the ticket.
Only issue with indentation.

SheenaSharma commented on structuredefinitions/England-MedicationRequest.xml:
MedicationRequest.medication[x]: Remove redundant code lines

This change needs to be done. Please remove the redundant code lines from MedicationRequest.medication[x].

KazeemHamzat replied:
Redundant codes have been removed.

HamzatKazeem committed:
Update England-MedicationRequest.

Remove redundant codelines for MedicationRequest.medication[x]
